### PR TITLE
Implement Custom RPC Methods

### PR DESCRIPTION
**Pull request by anvacaru**
This PR implements: 
- evm_snapshot
- evm_revert
- evm_increaseTime

Documentation for methods: https://github.com/trufflesuite/ganache-core#custom-methods

### CODE REVIEW

File: web3.md [hunk 1]
```diff
@@ -186,5 +193,37 @@ module WEB3
            <code> CODE </code>
            ...
          </account>
+
+    syntax KItem ::= "#evm_snapshot"
+ // --------------------------------
+    rule <k> #evm_snapshot => #pushNetworkState ~> #sendResponse( "result" : #unparseQuantity( size ( SNAPSHOTS ) ) ) ... </k>
+         <snapshots> SNAPSHOTS </snapshots>
```
Comment (ehildenb) [inline web3.md]: Is this supposed to send the number of snapshots before or after the `#pushNetworkState` operation?

File: web3.md [hunk 2]
```diff
@@ -186,5 +193,37 @@ module WEB3
...
+
+    syntax KItem ::= "#pushNetworkState"
+ // ------------------------------------
+    rule <k> #pushNetworkState => . ... </k>
+         <snapshots>... .List => ListItem (NETWORKSTATE) </snapshots>
```
Comment (ehildenb) [inline web3.md]: Format as `<snapshots> ... (.List => Listitem(NETWORKSTATE)) </snapshots>`

File: web3.md [hunk 3]
```diff
@@ -186,5 +193,37 @@ module WEB3
...
+
+    syntax KItem ::= "#evm_revert"
+ // ------------------------------
+    rule <k> #evm_revert => #sendResponse( "result" : "true" ) ... </k>
+         <params> [.JSONList] </params>
+         <snapshots>... ListItem(NETWORKSTATE) => .List </snapshots>
+         <network> _ => NETWORKSTATE </network>
+
+    rule <k> #evm_revert ... </k>
+         <params> [ (DATA => #parseHexWord(DATA))] </params>
```
Comment (ehildenb) [inline web3.md]: Formatting (add a space before the `]`)

File: web3.md [hunk 4]
```diff
@@ -186,5 +193,37 @@ module WEB3
...
+ // ------------------------------------
+    rule <k> #pushNetworkState => . ... </k>
+         <snapshots>... .List => ListItem (NETWORKSTATE) </snapshots>
+         <network> NETWORKSTATE </network>
+
+    syntax KItem ::= "#evm_revert"
+ // ------------------------------
+    rule <k> #evm_revert => #sendResponse( "result" : "true" ) ... </k>
+         <params> [.JSONList] </params>
+         <snapshots>... ListItem(NETWORKSTATE) => .List </snapshots>
```
Comment (ehildenb) [inline web3.md]: formatting (same as above)

File: web3.md [hunk 5]
```diff
@@ -186,5 +193,37 @@ module WEB3
...
+         <network> _ => NETWORKSTATE </network>
+
+    rule <k> #evm_revert ... </k>
+         <params> [ (DATA => #parseHexWord(DATA))] </params>
+
+    rule <k> #evm_revert => #sendResponse( "result" : "true" ) ... </k>
+         <params> [DATA, .JSONList ] </params>
```
Comment (ehildenb) [inline web3.md]: formatting (space after the `[`)

File: web3.md (hunk 1 above)
Comment (anvacaru) [reply web3.md]: It should return the `id` of the snapshot. So by returning the size of the snapshot list after `#pushNetworkState` I tried to send back the position of the newly added snapshot.

File: web3.md (hunk 5 above)
Comment (ehildenb) [reply web3.md]: Try `DATA:Int`

File: web3.md [hunk 6]
```diff
@@ -186,5 +193,37 @@ module WEB3
...
+         <snapshots> SNAPSHOTS => range(SNAPSHOTS, 0, DATA) </snapshots>
+         // <network> NETWORKSTATE => (SNAPSHOTS [DATA]) </network>
```
Comment (ehildenb) [inline web3.md]: Return type of `_[_]` is `KItem`. `{SNAPSHOTS [ DATA ]}:>NetworkCellFragment` (or `NetworkCell`).

**Pull request merged**